src/mqt/ionshuttler/linear/__init__.py

```
@@ -7,34 +7,55 @@
+
+def __getattr__(name: str) -> object:
+    """Resolve compiler entry points only when requested.
+
+    Returns:
+        The requested public object.
+
+    Raises:
+        AttributeError: If ``name`` is not a deferred public object.
+    """
+    if name == "LinearCompiler":
+        from mqt.ionshuttler.linear.compiler import (  # ruff: ignore[import-outside-top-level] - Deferred to keep the root import backend-neutral.
```

Ruff (0.16.2)

[warning] Invalid rule code in suppression: import-outside-top-level

Enable `lint.preview` to use rule names

(RUF102)